analyses/survival-analysis/survival-analysis_HGG_DMG.Rmd

```diff
@@ -0,0 +1,210 @@
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Make output directories.
+
+```{r Make output directories}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+Declare input file paths. 
+
+```{r Set input file paths}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+independent_primary_file <- file.path(data_dir, "independent-specimens.wgswxs.primary.tsv")
+```
+
+Declare output file paths for HGG/DMG all subtypes. 
+
+```{r Set output file paths}
+kap_meier_plot_file <- file.path(plots_dir, "survival_curve_HGG_DMG.pdf")
+kap_meier_model_file <- file.path(results_dir,"survival_model_HGG_DMG.tsv")
+```
+
+## Import the metadata
+
+```{r Read in metadata}
+# choose Diffuse astrocytic and oligodendroglial tumor samples
+# remove RNA-Seq because subtyping in HGG is from DNA abberations and samples where only RNA-Seq is available cannot be subtyped
+# remove cell-lines from this analysis as well
+metadata <- readr::read_tsv(metadata_file,
+                            col_types = readr::cols(molecular_subtype = readr::col_character())) %>%
+  filter(experimental_strategy !="RNA-Seq" & 
+           broad_histology=="Diffuse astrocytic and oligodendroglial tumor" &
+           broad_composition !="cell-line") %>% 
+  as.data.frame()
+
+# find patients where multiple subtypes are present 
+count_patient_subtype<-metadata %>% 
+  select(Kids_First_Participant_ID ,molecular_subtype) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID ) %>% 
+  summarise(count=n(),subtype=toString(molecular_subtype)) %>%
+  filter(count>1)
+
+# these patients might affect downstream analysis for survival and would need to be removed
+count_patient_subtype
+
+```
+
+#### Let's use independent_primary to use primary samples only and maybe will help filter cases where multiple subtypes are present per patient ID
+
+```{r Read in independent}
+independent_primary<-read_tsv(independent_primary_file)
+# select only independent primary bs_ids to get overall survival
+metadata<-metadata %>% filter(Kids_First_Biospecimen_ID %in% independent_primary$Kids_First_Biospecimen_ID)
+
+# find patients where multiple subtypes are present 
+count_patient_subtype<-metadata %>% 
+  select(Kids_First_Participant_ID ,molecular_subtype) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID ) %>% 
+  summarise(count=n(),subtype=toString(molecular_subtype)) %>%
+  filter(count>1)
+
+count_patient_subtype
+
+```
+#### Ok so no patients with multiple subtypes now.
+
+## Kaplan-Meier for all HGG/DMG subtypes
+
+The [Kaplan-Meier](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059453/) 
+survival curve shows the probability of survival over time.
+
+```{r Kaplan Meier}
+kap_fit <- survival_analysis(metadata,
+  ind_var = "molecular_subtype",
+  test = "kap.meier",
+  metadata_sample_col = "Kids_First_Biospecimen_ID"
+)
+# median and 0.95CI survival days
+kap_fit$model
```

Comment:
I'm not an expert in survival analyses, but my gut is telling me we should probably *not* be considering any N=1 here as there is no possible estimate of uncertainty - CI's are NA. @cansavvy, since you prepped the module here, any thoughts?

Reply:
I would only look at H3 K28 and H3 wildtype.

Reply:
I missed this comment, I'll implement this